tests/D2L.CodeStyle.Analyzers.Test/UnsafeStatics/UnsafeStaticsAnalyzerTests.cs

```diff
@@ -672,6 +674,61 @@ internal class Bar {
 			AssertSingleDiagnostic( test, 7, 41, "foo", "test.Tests.Foo" );
 		}
 
+		[Test]
+		public void DocumentWithStaticField_ReadonlyUnsafeBaseClassWithUnsafeInitializer_Diag() {
+			const string test = @"
+	using System;
+	namespace test {
+		class Tests {
+			interface IUnsafe { void Magic(); } // could be anythinggggggg
+
+			class Safe : IUnsafe {
+				void IUnsafe.Magic() {} // looks safe to me
+				public static readonly Safe Instance { get; } = new Safe();
+			}
+
+			private readonly static IUnsafe foo = Safe.Instance;
+		}
+	}";
+
+			// Safe isn't sealed
+			AssertSingleDiagnostic( test, 12, 36, "foo", "test.Tests.Safe" );

```

Comment:
This is actually safe, because we know that `Safe.Instance` is concretely `Safe` (doesn't need to be sealed). But that's one level down (inspect a field's declaration) that can come in another PR. It's good that there's a test for this though.